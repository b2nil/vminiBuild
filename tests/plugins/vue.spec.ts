import esbuild from "esbuild"
const { vuePlugin } = require('../plugin')

const compileVue = async (src: string) => {
  const res = await esbuild.build({
    entryPoints: [src],
    format: "esm",
    bundle: false,
    write: false,
    plugins: [vuePlugin()]
  })

  const out = res.outputFiles[0]
  const code = String.fromCharCode.apply(null, out.contents)

  return code
}

describe("vue plugin", () => {
  it("should transform .vue with normal script for bundling", async () => {
    const res = await compileVue("./tests/plugins/vue/src/script.vue")
    expect(res).toMatch(/import "\.\/components\/comp\.vue"/)
    expect(res).toMatch(/import "\.\/components\/comp2\.vue"/)
    expect(res).toMatch(/import "\.\/pseudo.wxs"/)
    expect(res).toMatch("function definePageConfig(config)")
    expect(res).toMatchSnapshot()
  })

  it("should transform .vue with script setup for bundling", async () => {
    const res = await compileVue("./tests/plugins/vue/src/scriptSetup.vue")
    expect(res).toMatch(/import "\.\/components\/comp\.vue"/)
    expect(res).toMatch(/import "\.\/components\/comp2\.vue"/)
    expect(res).toMatch(/import "\.\/test.wxs"/)
    expect(res).toMatch("function definePageConfig(config)")
    expect(res).toMatch("function defineProps(config)")
    expect(res).toMatch("function defineExpose(config)")
    expect(res).toMatch("function defineHookConfig(config)")

    expect(res).toMatchSnapshot()
  })
})

