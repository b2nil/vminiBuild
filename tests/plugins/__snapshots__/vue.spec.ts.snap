// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vue plugin should transform .vue with normal script for bundling 1`] = `
"import { ref } from \\"@vue-mini/wechat\\";
definePageConfig({
  a: { c: \\"d\\" },
  b: \\"a\\"
});
var script_default = definePage({
  components: {
    ComOne,
    ComTwo
  },
  props: { hi: Number },
  setup() {
    const data = ref(0);
    const method = () => {
      console.log();
    };
    return {
      data,
      method
    };
  }
});
function definePageConfig(config) {
  return config;
}
import \\"./components/comp.vue\\";
import \\"./components/comp2.vue\\";
import \\"./pseudo.wxs\\";
import \\"./pseudo.wxs\\";
export {
  script_default as default
};
"
`;

exports[`vue plugin should transform .vue with script setup for bundling 1`] = `
"import { ref, onPageScroll } from \\"@vue-mini/wechat\\";
definePageConfig({
  a: { c: \\"d\\" },
  b: \\"a\\"
});
defineProps({
  hi: Number
});
const data = ref(0);
const method = () => {
  console.log();
};
onPageScroll(({ scrollTop }) => {
  console.log(\\"scrollTop:\\", scrollTop);
});
defineExpose({
  data,
  method
});
defineHookConfig({
  listenPageScroll: true
});
function definePageConfig(config) {
  return config;
}
function defineProps(config) {
  return config;
}
function defineExpose(config) {
  return config;
}
function defineHookConfig(config) {
  return config;
}
const COMP_r3m0v3 = { ComOne, ComTwo };
var scriptSetup_default = COMP_r3m0v3;
import \\"./components/comp.vue\\";
import \\"./components/comp2.vue\\";
import \\"./test.wxs\\";
export {
  scriptSetup_default as default
};
"
`;
